main/src/ca/uwaterloo/flix/language/ast/OccurrenceAst.scala

```diff
@@ -210,6 +210,7 @@ object OccurrenceAst {
     case object ManyBranch extends Occur
     case object Dead extends Occur
     case object Many extends Occur
+    case object Sacred extends Occur
   }
```

Comment:
I think the above should be ordered:

Dead
Once
Many
ManyBranch
Sacred

and then you should add a comment to each, e.g.:

/**
 * Represents variables that occur exactly once in an expression.
 */
case class Once

Also, I would rename Sacred to `DontInline` etc.